src/linter.ts

```diff
@@ -176,15 +179,17 @@ class Linter {
     protected applyFixes(sourceFilePath: string, source: string, fixableFailures: RuleFailure[]): string {
         const fixesByFile = createMultiMap(fixableFailures, (f) => [f.getFileName(), f.getFix()!]);
         fixesByFile.forEach((fileFixes, filePath) => {
-            let fileNewSource: string;
+            let fileNewSource: string | undefined;
             if (path.resolve(filePath) === path.resolve(sourceFilePath)) {
                 source = Replacement.applyFixes(source, fileFixes);
                 fileNewSource = source;
-            } else {
+            } else if (!this.options.dryRun) {
                 const oldSource = fs.readFileSync(filePath, "utf-8");
                 fileNewSource = Replacement.applyFixes(oldSource, fileFixes);
             }
-            fs.writeFileSync(filePath, fileNewSource);
+            if (!this.options.dryRun && typeof fileNewSource === "string") {
```

Comment:
this basically reverts #2864 when `dryRun` is true. you don't want that

Reply:
`updateProgram` is called regardless of `dryRun`. Not sure what I'm missing?

Reply:
updateProgram reads the file from disk. With dryRun it is never updated

Reply:
Oh, wow. That's unfortunate. Know any way around that? I think using    `CompilerHost#getSourceFile` could work.